packages/react/src/__tests__/feature-app-container.test.tsx

```diff
@@ -548,16 +635,284 @@ describe('FeatureAppContainer', () => {
+          });
+
+          it('renders what children returns', async () => {
+            const children = jest.fn(({loading}) =>
+              loading ? 'is loading' : 'not loading'
+            );
+
+            const testRenderer = renderWithFeatureHubContext(
+              <FeatureAppContainer
+                featureAppId="testId"
+                featureAppDefinition={mockFeatureAppDefinition}
+                children={children}
+              />
+            );
+
+            expect(testRenderer.toJSON()).toBe('is loading');
+
+            await resolveLoadingPromise();
+
+            expect(testRenderer.toJSON()).toBe('not loading');
+          });
+        });
+
+        describe('when promise rejects', () => {
+          let mockError: Error;
+          beforeEach(() => {
+            mockError = new Error('loading promise rejected with error');
+          });
+          it('initially renders children with loading=true, then with loading=false', async () => {
+            const children = jest.fn(() => null);
+
+            renderWithFeatureHubContext(
+              <FeatureAppContainer
+                featureAppId="testId"
+                featureAppDefinition={mockFeatureAppDefinition}
+                children={children}
+              />
+            );
+
+            expect(children).toHaveBeenCalledTimes(1);
+            expect(children.mock.calls[0]).toMatchObject([{loading: true}]);
+
+            await rejectLoadingPromise(mockError);
+
+            expect(children).toHaveBeenCalledTimes(2);
+            expect(children.mock.calls[1]).toMatchObject([{loading: false}]);
+          });
+
+          it('calls children with rejection error', async () => {
+            const children = jest.fn(() => null);
+
+            renderWithFeatureHubContext(
+              <FeatureAppContainer
+                featureAppId="testId"
+                featureAppDefinition={mockFeatureAppDefinition}
+                children={children}
+              />
+            );
+
+            expect(children).toHaveBeenCalledTimes(1);
+            expect(children.mock.calls[0]).not.toMatchObject([
+              {error: mockError}
+            ]);
+
+            await rejectLoadingPromise(mockError);
+
+            expect(children).toHaveBeenCalledTimes(2);
+            expect(children.mock.calls[1]).toMatchObject([{error: mockError}]);
+          });
+
+          it(`doesn't pass featureAppNode to children after rejection`, async () => {
+            const children = jest.fn(() => null);
+
+            renderWithFeatureHubContext(
+              <FeatureAppContainer
+                featureAppId="testId"
+                featureAppDefinition={mockFeatureAppDefinition}
+                children={children}
+              />
+            );
+
+            expect(children).toHaveBeenCalledTimes(1);
+            expect(children.mock.calls[0]).toMatchObject([
+              {featureAppNode: expect.anything()}
```

Comment:
It's all written a bit weirdly here, because i had a hard time at getting the call parameter into the expect, because typescript is complaining, and i didn't want to snapshot the whole object inside an array in every test.

I will try to restructure this again, and maybe put a `@ts-ignore` before errors that are checked against via expects already, for readability and being more exact.

Reply:
found out that i only have to type the parameter of the mock function for typescript to be happy!